src/core/unit.c

```diff
@@ -3316,12 +3316,16 @@ int unit_set_default_slice(Unit *u) {
                 if (MANAGER_IS_SYSTEM(u->manager))
                         slice_name = strjoina("system-", escaped, ".slice");
                 else
-                        slice_name = strjoina(escaped, ".slice");
-        } else
-                slice_name =
-                        MANAGER_IS_SYSTEM(u->manager) && !unit_has_name(u, SPECIAL_INIT_SCOPE)
-                        ? SPECIAL_SYSTEM_SLICE
-                        : SPECIAL_ROOT_SLICE;
+                        slice_name = strjoina("app-", escaped, ".slice");
+        } else {
+                if (MANAGER_IS_SYSTEM(u->manager))
+                        slice_name =
+                                !unit_has_name(u, SPECIAL_INIT_SCOPE)
+                                ? SPECIAL_SYSTEM_SLICE
+                                : SPECIAL_ROOT_SLICE;
```

Comment:
inverted conditions are weird. when you touch this maybe swap the order of the ternary op here, i.e. 

```c
                        slice_name =
                                unit_has_name(u, SPECIAL_INIT_SCOPE)
                                ? SPECIAL_ROOT_SLICE
                                : SPECIAL_SYSTEM_SLICE;
```